src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/WindowsForms/MyAppDocument.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Xml.Linq;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.WindowsForms;
+
+internal class MyAppDocument
+{
+    private readonly XDocument doc;
+    private readonly XElement root;
+
+    private readonly string name;
+
+    public MyAppDocument(string fileName, string filePath)
+    {
+        var xmlString = File.ReadAllText(filePath);
+        doc = XDocument.Parse(xmlString);
```

Comment:
```suggestion
        doc = XDocument.Load(filePath);
```

This allows `XDocument` to parse the file as it streams from disk, rather than allocating its entire contents in a string before parsing.